src/pseudopeople/loader.py

```diff
@@ -35,11 +35,12 @@ def load_and_prep_1040_data(data_path: dict, user_filters: List[Tuple]) -> pd.Da
     # Load data
     df_1040 = load_standard_dataset_file(data_path[DatasetNames.TAXES_1040], user_filters)
     # We do not want to filter by state for dependents
-    for i in range(len(user_filters)):
-        if user_filters[i][0] == COLUMNS.mailing_state.name:
-            user_filters.pop(i)
+    no_state_user_filters = user_filters.copy()
+    for i in range(len(no_state_user_filters)):
+        if no_state_user_filters[i][0] == COLUMNS.mailing_state.name:
+            no_state_user_filters.pop(i)
```

Comment:
You didn't solve the issue that Zeb raised. The issue is not so much that you are modifying the previously created filters, it's that you're modifying an object while you are iterating over it.
```
In [1]: a = [2, 3, 1, 2]

In [2]: for i in range(len(a)):
   ...:     if a[i] == 3:
   ...:         print("this worked")
   ...:
this worked

In [3]: for i in range(len(a)):
   ...:     if a[i] == 3:
   ...:         a.pop(i)
   ...:
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[9], line 2
      1 for i in range(len(a)):
----> 2     if a[i] == 3:
      3         a.pop(i)

IndexError: list index out of range
```

I'd suggest using Zeb's proposed solution, unless that doesn't work for some reason.